src/decorators/SequentialEventListener.ts

```diff
@@ -1,21 +1,16 @@
-import type { IEventConstructor } from '../ISequentialEventListener';
+import type { IEvent } from '../ISequentialEventListener';
 import 'reflect-metadata';
+import { randomUUID } from 'crypto';
+import { SEQUENTIAL_EVENT, SEQUENTIAL_EVENT_LISTENER } from '../constants';
 
-export const SEQUENTIAL_EVENT_LISTENER = '__sequentialEventListener';
-
-export const SequentialEventListener = (event: IEventConstructor): ClassDecorator => {
+export const SequentialEventListener = (...events: IEvent[]): ClassDecorator => {
   return (target: object) => {
-    Reflect.defineMetadata(SEQUENTIAL_EVENT_LISTENER, event, target);
-  };
-};
+    events.forEach((event) => {
+      if (!Reflect.hasOwnMetadata(SEQUENTIAL_EVENT, event)) {
+        Reflect.defineMetadata(SEQUENTIAL_EVENT, { id: randomUUID() }, event);
```

Comment:
I don't understand how it works, especially with the line 14.
Here it only will assign one event basically.

Please create an e2e test with nest application to check that it is working as it shold.

Reply:
refactored the code so all metadata changes are now handled in `EventMetadata` and `EventListenerMetadata`